includes/bootstrap.php

```diff
@@ -0,0 +1,179 @@
+	if ( ! is_admin() ) {
+		return;
+	}
+	?>
+	<div class="notice notice-error">
+		<p><?php echo esc_html( $message ); ?></p>
+	</div>
+	<?php
+}
+
+/**
+ * Checks if the PHP version meets the minimum requirement.
+ *
+ * @since 0.1.0
+ *
+ * @return bool True if PHP version is sufficient, false otherwise.
+ */
+function check_php_version(): bool {
+	if ( version_compare( phpversion(), AI_MIN_PHP_VERSION, '<' ) ) {
+		add_action(
+			'admin_notices',
+			function () {
+				version_notice(
+					sprintf(
+						/* translators: 1: Required PHP version, 2: Current PHP version */
+						__( 'AI plugin requires PHP version %1$s or higher. You are running PHP version %2$s.', 'ai' ),
+						AI_MIN_PHP_VERSION,
+						PHP_VERSION
+					)
+				);
+			}
+		);
+		return false;
+	}
+	return true;
+}
+
+/**
+ * Checks if the WordPress version meets the minimum requirement.
+ *
+ * @since 0.1.0
+ *
+ * @return bool True if WordPress version is sufficient, false otherwise.
+ */
+function check_wp_version(): bool {
+	global $wp_version;
+
+	if ( version_compare( $wp_version, AI_MIN_WP_VERSION, '<' ) ) {
+		add_action(
+			'admin_notices',
+			function () {
+				global $wp_version;
+				version_notice(
+					sprintf(
+						/* translators: 1: Required WordPress version, 2: Current WordPress version */
+						__( 'AI plugin requires WordPress version %1$s or higher. You are running WordPress version %2$s.', 'ai' ),
+						AI_MIN_WP_VERSION,
+						$wp_version
+					)
+				);
+			}
+		);
+		return false;
+	}
+	return true;
+}
+
+/**
+ * Displays admin notice about missing Composer autoload files.
+ *
+ * @since 0.1.0
+ */
+function display_composer_notice(): void {
+	?>
+	<div class="notice notice-error">
+		<p>
+			<?php
+			printf(
+				/* translators: %s: composer install command */
+				esc_html__( 'Your installation of the AI plugin is incomplete. Please run %s.', 'ai' ),
+				'<code>composer install</code>'
+			);
+			?>
+		</p>
+	</div>
+	<?php
+}
+
+/**
+ * Loads the plugin after checking requirements.
+ *
+ * @since 0.1.0
+ */
+function load(): void {
+	static $loaded = false;
+
+	// Prevent loading twice.
+	if ( $loaded ) {
+		return;
+	}
+
+	// Check version requirements.
+	if ( ! check_php_version() || ! check_wp_version() ) {
+		return;
+	}
+
+	// Load Composer autoloader.
+	if ( ! file_exists( AI_PLUGIN_DIR . 'vendor/autoload.php' ) ) {
+		add_action( 'admin_notices', __NAMESPACE__ . '\display_composer_notice' );
+		return;
+	}
+	require_once AI_PLUGIN_DIR . 'vendor/autoload.php';
+
+	$loaded = true;
+
+	// Initialize plugin.
+	try {
+		$registry = new Feature_Registry();
+		$loader   = new Feature_Loader( $registry );
+		$loader->register_default_features();
+		$loader->initialize_features();
+	} catch ( \Exception $e ) {
+		_doing_it_wrong(
+			__NAMESPACE__ . '\load',
+			sprintf(
+				/* translators: %s: Error message. */
+				esc_html__( 'AI Plugin initialization failed: %s', 'ai' ),
+				esc_html( $e->getMessage() )
+			),
+			'0.1.0'
+		);
+	}
```

Comment:
+1 in regards to @mindctrl's comment, this should probably only happen on `init`:
* Only before `init` the user gets initialized by default, and that influences translation strings. It's a best practice to not run any translation functions before the user is normally initialized.
* It's common in WordPress for things to be registered on `init`, so this would tie in nicely with that practice.
* It would enforce that no feature can break these patterns either. `init` is sufficiently early in the WordPress bootstrap process for probably any kind of interaction a specific feature may need to support.